### PR TITLE
Support for Pantheon specific RPC calls

### PR DESCRIPTION
**Pull request by benesjan**
### What does this PR do?
*Adds support for Pantheon specific RPC calls*

### Where should the reviewer start?
*Pantheon.java interface*

There are missing tests for clicque related methods. We didn't implement them on time because we managed to make clicque work 5 minutes before the hackathon presentations (there is a bug in a clicque docker image mentioned in the Pantheon documentation).

### CODE REVIEW

**Comment by benesjan** [comment]
Hi, if there are any issues with the code just let me know and I'll fix them.

File: pantheon/src/main/java/org/web3j/protocol/pantheon/JsonRpc2_0Pantheon.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    public Request<?, PantheonEthAccountsMapResponse> cliqueProposals() {
+        return new Request<>(
+                "clique_proposals",
+                Collections.<String>emptyList(),
+                web3jService,
+                PantheonEthAccountsMapResponse.class);
+    }
+
+    @Override
+    public Request<?, PantheonFullDebugTraceResponse> debugTraceTransaction(String transactionHash, Map<String, Boolean> options) {
```
Comment (iikirilov) [inline pantheon/src/main/java/org/web3j/protocol/pantheon/JsonRpc2_0Pantheon.java]: Geth call for this exists but is not currently implemented in geth module [see here](https://github.com/ethereum/go-ethereum/wiki/Management-APIs#debug_tracetransaction)

**Comment by conor10** [comment]
Hey guys - let’s leave the duplication in for now - it’s small and what I suggested at the hackathon

**Comment by iikirilov** [comment]
Ok - I am ok with that - let's fix up the checkstyle errors and we can merge

```
:pantheon:checkstyleMain[ant:checkstyle] [ERROR] /home/ivaylo/dev/playground/web3j/pantheon/src/main/java/org/web3j/protocol/pantheon/JsonRpc2_0Pantheon.java:86: Line is longer than 100 characters (found 131). [LineLength]
[ant:checkstyle] [ERROR] /home/ivaylo/dev/playground/web3j/pantheon/src/main/java/org/web3j/protocol/pantheon/Pantheon.java:13: Import statement for 'java.util.Map' is in the wrong order. Should be in the 'STANDARD_JAVA_PACKAGE' group, expecting not assigned imports on this line. [CustomImportOrder]
[ant:checkstyle] [ERROR] /home/ivaylo/dev/playground/web3j/pantheon/src/main/java/org/web3j/protocol/pantheon/Pantheon.java:34: Line is longer than 100 characters (found 123). [LineLength]
[ant:checkstyle] [ERROR] /home/ivaylo/dev/playground/web3j/pantheon/src/main/java/org/web3j/protocol/pantheon/response/FullDebugTraceInfo.java:12: Line is longer than 100 characters (found 105). [LineLength]
[ant:checkstyle] [ERROR] /home/ivaylo/dev/playground/web3j/pantheon/src/main/java/org/web3j/protocol/pantheon/response/FullDebugTraceInfo.java:23:26: '{' at column 26 should have line break after. [LeftCurly]
[ant:checkstyle] [ERROR] /home/ivaylo/dev/playground/web3j/pantheon/src/main/java/org/web3j/protocol/pantheon/response/FullDebugTraceInfo.java:23:44: '}' at column 44 should be alone on a line. [RightCurly]
[ant:checkstyle] [ERROR] /home/ivaylo/dev/playground/web3j/pantheon/src/main/java/org/web3j/protocol/pantheon/response/FullDebugTraceInfo.java:33:36: '{' at column 36 should have line break after. [LeftCurly]
[ant:checkstyle] [ERROR] /home/ivaylo/dev/playground/web3j/pantheon/src/main/java/org/web3j/protocol/pantheon/response/FullDebugTraceInfo.java:33:58: '}' at column 58 should be alone on a line. [RightCurly]
[ant:checkstyle] [ERROR] /home/ivaylo/dev/playground/web3j/pantheon/src/main/java/org/web3j/protocol/pantheon/response/FullDebugTraceInfo.java:35:34: '{' at column 34 should have line break after. [LeftCurly]
[ant:checkstyle] [ERROR] /home/ivaylo/dev/playground/web3j/pantheon/src/main/java/org/web3j/protocol/pantheon/response/FullDebugTraceInfo.java:35:68: '}' at column 68 should be alone on a line. [RightCurly]
[ant:checkstyle] [ERROR] /home/ivaylo/dev/playground/web3j/pantheon/src/main/java/org/web3j/protocol/pantheon/response/FullDebugTraceInfo.java:41:33: '{' at column 33 should have line break after. [LeftCurly]
[ant:checkstyle] [ERROR] /home/ivaylo/dev/playground/web3j/pantheon/src/main/java/org/web3j/protocol/pantheon/response/FullDebugTraceInfo.java:41:65: '}' at column 65 should be alone on a line. [RightCurly]
[ant:checkstyle] [ERROR] /home/ivaylo/dev/playground/web3j/pantheon/src/main/java/org/web3j/protocol/pantheon/response/StructLogs.java:18: Line is longer than 100 characters (found 149). [LineLength]
 FAILED
```

You can run the checkstyle again with `./gradlew check` from the root directory

File: pantheon/src/main/java/org/web3j/protocol/pantheon/response/StructLogs.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+public class StructLogs {
+
+    private int pc;
+    private String op;
+    private int gas;
+    private int gasCost;
+    private int depth;
+    private List<String> stack;
+    private List<String> memory;
+    private Map<BigInteger, String> storage; // maybe List<Map<BigInteger, String>>
```
Comment (lucassaldanha) [inline pantheon/src/main/java/org/web3j/protocol/pantheon/response/StructLogs.java]: Any need to keep this comment here?

**Comment by lucassaldanha** [comment]
It would be good to add some tests around this code. The RequestTester and ResponseTester can help to make sure serialisation is doing what it is expected.

File: pantheon/src/main/java/org/web3j/protocol/pantheon/response/StructLogs.java (hunk 2 above)
Comment (benesjan) [reply pantheon/src/main/java/org/web3j/protocol/pantheon/response/StructLogs.java]: There is definitely no need for that. I'll remove it in the next commit.

**Comment by conor10** [comment]
Once we have ResponseTests as per @lucassaldanha's comment we're good to merge

**Comment by benesjan** [comment]
I implemented the response tests for 2 methods: clicqueGetSigners and cliqueProposals. Another method which would be worthy of testing is debugTraceTransaction. However the documentation for the method doesn't contain example response and I didn't manage to call the method successfully using RPC on Rinkeby testnet so I am not sure in what state is the implementation of the method in Pantheon.

RPC CALL:
`curl -X POST --data '{"jsonrpc":"2.0","method":"debug_traceTransaction","params”:[“0x2dd50756cfd189177d0de4d65b7926459d7bb01de197c26517d05cd2b1c77ebc”,{“disableStorage":true}],"id”:1}' http://localhost:8545/
`

RESPONSE:
`{
  "jsonrpc" : "2.0",
  "id" : null,
  "error" : {
    "code" : -32600,
    "message" : "Invalid Request"
  }
`

Other methods  return either trivial responses (BooleanResponse) or responses which are already tested elsewhere so I don't think there is a need to test them.

**Pull request merged**